### ISSUE TITLE
liveReloadBaseUrl with absolute url no longer supported

### ISSUE THREAD

**Issue by wongpeiyi** [ember-cli/ember-cli-inject-live-reload#33]
In the past, we could use `"liveReloadBaseUrl": "http://localhost:4200/"` which is necessary for using ember-cli-deploy for a development workflow, i.e. accessing from `http://localhost:3000/`

However, with https://github.com/rwjblue/ember-cli-inject-live-reload/commit/3473a6479d65a73097a0ddb2e4298b714ed3cfad this no longer seems to work

Should this be

``` javascript
app.use((options.rootURL || options.baseURL) + 'ember-cli-live-reload.js', // ...
```

instead?

**Comment by jgoclawski** [comment]
Hi, I'm the author of the mentioned commit. 
There was a refactor before that commit (part of PR #28) and if you take a look at the [full index.js file](https://github.com/rwjblue/ember-cli-inject-live-reload/blob/3473a6479d65a73097a0ddb2e4298b714ed3cfad/index.js), you will see that:
`var baseURL = options.liveReloadBaseUrl || options.rootURL || options.baseURL;`

So if you set `liveReloadBaseUrl` if will be used. Do you say it **shouldn't** be used?

You said that it no longer seems to work. What are the symptoms? Do you get any errors or any unexpected output in HTML?

**Comment by wongpeiyi** [comment]
Thanks for the response @jgoclawski 

To clarify, I'm using the [ember-cli-deploy development workflow](http://ember-cli-deploy.com/docs/v0.6.x/development-workflow/) – index.html served on localhost:3000, ember server running on localhost:4200. 

Previously (before your PR), using `liveReloadBaseUrl: "http://localhost:4200/"` would:

1. Inject `<script src="http://localhost:4200/ember-cli-live-reload.js" type="text/javascript">` (when accessing from localhost:3000)

2. Because of `app.use(options.baseURL + 'ember-cli-live-reload.js'`, serve the javascript via a relative path, i.e. `app.use('/ember-cli-live-reload.js'`

This allowed livereload to be served on http://localhost:4200/ember-cli-live-reload.js even when accessing from http://localhost:3000/index.html

With this change to use `options.liveReloadBaseUrl`, using absolute paths in `liveReloadBaseUrl` no longer work, because AFAIK `app.use` doesn't intercept absolute paths, i.e. point (2) above becomes `app.use('http://localhost:4200/ember-cli-live-reload.js'`

> What are the symptoms? 

The script is injected but I get a 404 for ember-cli-live-reload.js (because it's not being intercepted)

>  Do you say it shouldn't be used?

Perhaps using absolute paths in `liveReloadBaseUrl` was never intended and it worked by mistake in the past. If it's intended to work, perhaps `liveReloadBaseUrl` should be converted to a relative path before `app.use`?

Let me know if this makes sense? If needed I can try creating a repo to illustrate

**Comment by lukemelia** [comment]
Would https://github.com/ember-cli/ember-cli-inject-live-reload/pull/36 solve your problem?

**Comment by wongpeiyi** [comment]
@lukemelia yes, thanks -- tested and it works

**Comment by pieter-v** [comment]
@lukemelia test failed because now we have the same problem as #35

**Comment by wongpeiyi** [comment]
@pieter-v I did face above as well, but I think it's a separate issue, and can be dealt with in the meantime by defining `"liveReloadOptions"` in .ember-cli

**Comment by pieter-v** [comment]
@wongpeiyi  thanks specifying the 'liveReloadOptions' solved the problem

**Comment by rwjblue** [comment]
Closing (seems like #25, #36 and #40 should have fixed the various issues being discussed).

**Comment by aprescott** [comment]
It looks like this was closed because (partially) #36 addresses the problem, however #36 is still open. Since #36 isn't merged/released yet, is there any way around this issue?

**Comment by aprescott** [comment]
Ah, I misread a comment above. Pinning to 1.4.0 gets this working with `liveReloadBaseUrl` set appropriately. It'd be great to run the latest release and still have this functionality though!
